doc/arch/adr-001-notify-integration.md

```diff
@@ -0,0 +1,424 @@
+  responsibility to consolidate these
+- Allow for multiple entries with same ContentChange and Subscription - to
+  allow a content change to be reprocessed in the event of an error
+- Support Email records being removed from the system for auditing process
+- Be deleteable for a scenario that a user changes a subscription before an
+  email is sent
+
+#### To determine
+
+- What is the archiving strategy on these? Will they be persisted forever or
+  match persistence strategy of Emails for instance?
+
+### Email
+
+**Responsibility**: A distinct email that the system will be sending or has sent
+
+#### Shall
+
+- Be created any time Email Alert API sends an email
+- Store information such as
+  - Recipient
+  - Subject
+  - Body (in Notify Markdown)
+- Delivery status
+
+#### Shall Not
+
+- Have more than one recipient
+- Know anything about the content that led to its generation
+
+#### Should
+
+- Only exist in the database for a non permanent time period
+- Be associated with a Subscriber for logging purposes
+
+#### May
+
+- Be archived into a log somewhere for later analysis
+
+### DeliveryAttempt
+
+**Responsibility**: Storing the information used to communicate with the
+transportation service for email
+
+#### Shall
+
+- Store information such as
+  - Email associated with
+  - Provider (e.g. Notify)
+  - Provider Reference (An id to look up the email)
+  - Status (success/pending/failure)
+- Any provider specific errors/warnings
+
+#### Should
+
+- Be created whenever an email is attempted to be sent
+- Be updated when sufficient information is known about whether an email has
+  sent or not
+- Be deleted when an Email is deleted
+
+#### May
+
+- Be created in a success state if it was known the delivery attempt was
+  successful synchronously
+- Be used in future with different transport mechanisms if we expand beyond
+  Email
+
+### ContentChangeController
+
+**Responsibility**: Storing a ContentChange, enquing that for processing,
+returning a success
+
+#### Shall
+
+- Create a ContentChange model
+- Enqueue a SubscriptionContentWorker job
+
+#### To determine
+
+- Who has responsibility for the Worker failing to run?
+
+### SubscriptionContentWorker
+
+**Responsibility**: Responsible for determining which subscriptions should receive
+an email prospectively and creating SubscriptionContent objects accordingly
+
+#### Shall
+
+- Be given sufficient input to lookup a ContentChange
+- Liaise with SubscriptionMatcher to determine all subscriptions that match a
+  ContentChange
+- Create SubscriptionContent entities for all subscriptions that match a
+  ContentChange
+- Trigger EmailGenerationWorker for emails that have to run immediately
+
+#### Should not
+
+- Be able to have multiple instances of this worker processing the same
+  ContentChange concurrently
+
+#### May
+
+- Be passed in frequency as an argument to match subscriptions of a given
+  frequency, as to allow rebuilding of emails that went out in error
```

Comment:
nit - s/as/in order 
or just remove it. The comma is potentially not needed either.